src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c

```diff
@@ -27,63 +67,661 @@ const ActorInit Obj_Oshihiki_InitVars = {
+        highestFloor = temp;
+    } else if ((this->floorBgIds[temp] == 0x32 /*BGCHECK_SCENE*/) &&
+               ((phi_f0 - this->floorHeights[highestFloor]) > -0.001f)) {
+        highestFloor = temp;
+    }
+    if (this->floorHeights[temp + 1] > this->floorHeights[highestFloor]) {
+        highestFloor = temp + 1;
+    } else if ((this->floorBgIds[temp + 1] == 0x32 /*BGCHECK_SCENE*/) &&
+               ((this->floorHeights[temp + 1] - this->floorHeights[highestFloor]) > -0.001f)) {
+        highestFloor = temp + 1;
+    }
+    if (this->floorHeights[temp + 2] > this->floorHeights[highestFloor]) {
+        highestFloor = temp + 2;
+    } else if ((this->floorBgIds[temp + 2] == 0x32 /*BGCHECK_SCENE*/) &&
+               ((this->floorHeights[temp + 2] - this->floorHeights[highestFloor]) > -0.001f)) {
+        highestFloor = temp + 2;
+    }
+    if (this->floorHeights[temp + 3] > this->floorHeights[highestFloor]) {
+        highestFloor = temp + 3;
+    } else if ((this->floorBgIds[temp + 3] == 0x32 /*BGCHECK_SCENE*/) &&
+               ((this->floorHeights[temp + 3] - this->floorHeights[highestFloor]) > -0.001f)) {
+        highestFloor = temp + 3;
+    }
+    return highestFloor;
+}
+
+void ObjOshihiki_SetGround(ObjOshihiki* this, GlobalContext* globalCtx) {
+    ObjOshihiki_ResetFloors(this);
+    ObjOshihiki_SetFloors(this, globalCtx);
+    this->highestFloor = ObjOshihiki_GetHighestFloor(this);
+    this->dyna.actor.groundY = this->floorHeights[this->highestFloor];
+}
+
+s32 ObjOshihiki_CheckFloor(ObjOshihiki* this, GlobalContext* globalCtx) {
+    ObjOshihiki_SetGround(this, globalCtx);
+
+    if ((this->dyna.actor.groundY - this->dyna.actor.posRot.pos.y) >= -0.001f) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.groundY;
+        return 1;
+    }
+
+    return 0;
+}
+
+s32 ObjOshihiki_CheckGround(ObjOshihiki* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.posRot.pos.y <= -31990.0f) {
+        // Warning : Push-pull block fell too much
+        osSyncPrintf("Warning : 押し引きブロック落ちすぎた(%s %d)(arg_data 0x%04x)\n", "../z_obj_oshihiki.c", 809,
+                     this->dyna.actor.params);
+        Actor_Kill(&this->dyna.actor);
+        return 0;
+    }
+    if ((this->dyna.actor.groundY - this->dyna.actor.posRot.pos.y) >= -0.001f) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.groundY;
+        return 1;
+    }
+    return 0;
+}
+
+s32 ObjOshihiki_CheckWall(GlobalContext* globalCtx, s16 angle, f32 direction, ObjOshihiki* this) {
+    f32 maxDist = ((direction >= 0.0f) ? 1.0f : -1.0f) * (300.0f * this->dyna.actor.scale.x + 20.0f - 0.5f);
+    f32 sn = Math_Sins(angle);
+    f32 cs = Math_Coss(angle);
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        Vec3f faceVtx;
+        Vec3f faceVtxNext;
+        Vec3f posResult; // dummy value
+        Vec3f faceVtxOffset;
+        u32 bgId;               // dummy value
+        CollisionPoly* outPoly; // dummy value
```

Comment:
Imo there is no need to specify that the function doesn't end up reading these values. Or at least if you want to specify it, it should probably be part of the variable name. But I think it's better to just leave it available so whoever works on this function later can make use of them without having to rename them.
```suggestion
        Vec3f posResult;
        Vec3f faceVtxOffset;
        u32 bgId;
        CollisionPoly* outPoly;
```